### DISCUSSION TITLE
in lightning,what's the function of the lightning_loftr.py?

### DISCUSSION THREAD

**Discussion by zhubingerdai** [zju3dv/LoFTR#108] [Q&A]
I didn't see lightning Loftr in the paper, I don't know what lightning and Lightning_loftr are here?

**Comment by zehongs** [comment]
Hi, the lightning is a training framework based on pytorch. Please check https://www.pytorchlightning.ai/ for more information.

**Reply by zhubingerdai** [reply]
thank you。What does geometry.py do that is also not mentioned in the article？

**Reply by zhubingerdai** [reply]
I want to ask you one more question.I read the article and looked at the code and didn't quite understand superglgly.py.I want to know in method 1/2/3/4, which method is it used, and what is its role?

**Reply by zehongs** [reply]
1. geometry.py contains basic operation of projective geometry.
2. buggy implementation.
3. the ablation study (Table 6) evaluates variants of LoFTR, not superglue.

**Comment by zhubingerdai** [comment]
I would like to know is Scannet data set all used for training? About 1.2 TB?This data set is hard to download。
andHave you tried running it on a 2080TI, 4GPUs? I'm a little worried about the configuration of the lab computers。。。。。

**Reply by zehongs** [reply]
I would suggest you to use the MegaDepth, which gives better depth-maps and is smaller. 
You might need more GPUs, but 4 GPUs will also work.

**Reply by zhubingerdai** [reply]
ok，thank you ！
I downloaded MegaDepth

**Reply by HUSTNO1WXY** [reply]
Can one gpu also work?

**Reply by zehongs** [reply]
Probably not...

**Comment by zhubingerdai** [comment]
e。。I have two questions. 1.Where is the MNN algorithm defined? Or in which module? 2.What role does warp kpts of geometry.py in supervision？

**Reply by zehongs** [reply]
1. coarse_matching.py
2. to find correspondence?

**Comment by zhubingerdai** [comment]


**Reply by zehongs** [reply]
I would suggest debugging with pdb to find if those paths of files are resolved correctly.
I think this is not a common issue, so we won't look into this. Hope you can understand!

**Reply by HUSTNO1WXY** [reply]
I think I also come up with this problem, how do you solve it?

**Comment by zhubingerdai** [comment]
There is no solution. I wonder if my Undistorted_SfM file has not been downloaded completely










At 2021-12-12 15:35:22, "Dongyue Li" ***@***.***> wrote:

I think I also come up with this problem, how do you solve it?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
